services/brig/src/Brig/Calling/API.hs

```diff
@@ -103,34 +110,37 @@ getCallsConfig _ _ = do
         (Public.rtcConfIceServers . traverse . Public.iceURLs . traverse . Public.turiTransport)
         Nothing
 
-newConfig :: MonadIO m => TURN.Env -> Maybe (Range 1 10 Int) -> m Public.RTCConfiguration
-newConfig env limit = do
+newConfig :: MonadIO m => Calling.Env -> Maybe SFTEnv -> Maybe (Range 1 10 Int) -> m Public.RTCConfiguration
+newConfig env mSftEnv limit = do
   let (sha, secret, tTTL, cTTL, prng) = (env ^. turnSHA512, env ^. turnSecret, env ^. turnTokenTTL, env ^. turnConfigTTL, env ^. turnPrng)
   -- randomize list of servers (before limiting the list, to ensure not always the same servers are chosen if limit is set)
-  randomizedUris <- liftIO $ randomize (env ^. turnServers)
+  randomizedUris <- liftIO $ randomize (List1.toNonEmpty $ env ^. turnServers)
   let limitedUris = case limit of
         Nothing -> randomizedUris
         Just lim -> limitedList randomizedUris lim
   -- randomize again (as limitedList partially re-orders uris)
   finalUris <- liftIO $ randomize limitedUris
   srvs <- for finalUris $ \uri -> do
     u <- liftIO $ genUsername tTTL prng
-    pure $ Public.rtcIceServer (List1.singleton uri) u (computeCred sha secret u)
-  pure $ Public.rtcConfiguration srvs cTTL
+    pure $ Public.rtcIceServer (uri :| []) u (computeCred sha secret u)
+  sftSrvEntries <- maybe (pure Nothing) (readIORef . sftServers) mSftEnv
+  -- According to RFC2782, the SRV Entries are supposed to be tried in order of
+  -- priority and weight, but the clients try all of them concurrently, so there
```

Comment:
I thought we had agreed in https://github.com/zinfra/backend-issues/issues/1453#issue-621270722 as well as https://github.com/zinfra/backend-issues/issues/1453#issuecomment-663070245 to randomize the SFT server list? Client behaviour may change, and even if all of them are currently tried concurrently, the behaviour may in the future only try the first two. In addition, there is no real concurrency, and some servers in the list will "win" by returning the request first. Since this is the only place to control the distribution of requests to servers, i.e. "load balancing", I'm still in favour of randomizing the list here.

Reply:
Implemented in 009961d852839e28bb5b6844a0399775025dbe64